Lib/zlib.py

```diff
@@ -224,10 +228,20 @@ def decompress(self, string, max_length=0):
 
         self.inflater.setInput(string)
         inflated = _get_inflate_data(self.inflater, max_length)
+        self._crc32.update(inflated)
 
         r = self.inflater.getRemaining()
         if r:
-            if max_length and not self.inflater.finished():
+            if self.gzip and self.inflater.finished() and r == 8:
```

Comment:
If we don't also strip off the trailer and leave it in unused_data, programs that expect the stream to be emptied keep resubmitting unused  data which just throws a warning about not enough bytes to decode until a recursion limit is exceeded.  This does one better by validating the content size and crc.